network/src/noise/handshake.rs

```diff
@@ -323,6 +324,12 @@ impl NoiseUpgrader {
 
             // store the timestamp
             anti_replay_timestamps.store_timestamp(their_public_key, client_timestamp);
+        } else if payload.len() != 0 {
+            // TODO: security logging
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidData,
+                "noise: a handshake payload was received in the full-node network: {}",
+            ));
```

Comment:
I wonder if we already get this logging via the error same as above...

Reply:
yeah, and in any case it'll be cleaner to consolidate the logging on the caller side :o good point

Reply:
1. probably remove the "{}" if you're not going to format anything : p

2. maybe call this "noise: received an unexpected handshake payload on a public-facing full-node network"?

Reply:
> probably remove the "{}" if you're not going to format anything : p

🤦🏼‍♂️